app/src/components/modal/modal_deposit_withdraw/index.tsx

```diff
@@ -374,41 +410,98 @@ export const ModalDepositWithdraw = (props: Props) => {
+                    ? `${formatBigNumber(minDaiExchange, STANDARD_DECIMALS, 2)} DAI`
+                    : `${formatBigNumber(minOmenExchange, omenToken.decimals, 2)} OMN`}
+                </span>
+              </ExchangeDataItem>
+              <ExchangeDataItem style={{ marginTop: '12px' }}>
+                <div style={{ display: 'flex' }}>
+                  {exchangeType === ExchangeType.withdraw ? 'Withdraw' : 'Deposit'} Fee
+                  <div
+                    data-arrow-color="transparent"
+                    data-for="feeInfo"
+                    data-tip={`Bridge Fee ${
+                      currencySelected === ExchangeCurrency.Omen && exchangeType === ExchangeType.withdraw
+                        ? '0.10%'
+                        : '0.00%'
+                    }`}
+                  >
+                    <IconInfo hasCircle style={{ marginLeft: '8px' }} />
+                  </div>
+                </div>
+
+                <span>
+                  {currencySelected === ExchangeCurrency.Dai
+                    ? '0.00 DAI'
+                    : exchangeType === ExchangeType.withdraw
+                    ? `${formatNumber(formatBigNumber(displayFundAmount.div(1000), omenToken.decimals, 3), 2)} OMN`
+                    : '0.00 OMN'}
+                </span>
+              </ExchangeDataItem>
+              <Divider />
+              <ExchangeDataItem>
+                <span>Total</span>
+                <span>
+                  {currencySelected === ExchangeCurrency.Omen && exchangeType === ExchangeType.withdraw
+                    ? `${formatBigNumber(
+                        displayFundAmount.sub(displayFundAmount.div(1000)),
+                        omenToken.decimals,
+                        3,
+                      )} OMN`
+                    : `${formatBigNumber(displayFundAmount, STANDARD_DECIMALS, 2)} ${
+                        currencySelected === ExchangeCurrency.Omen ? 'OMN' : 'DAI'
+                      }`}
+                </span>
+              </ExchangeDataItem>
+            </>
           )}
-          <DepositWithdrawButton
-            buttonType={ButtonType.primaryAlternative}
-            disabled={isDepositWithdrawDisabled}
-            onClick={depositWithdraw}
-          >
-            {exchangeType}
-          </DepositWithdrawButton>
+
+          <BottomButtons>
+            {(currencySelected === ExchangeCurrency.Dai ? daiWalletAllowance : omenWalletAllowance) && (
+              <ApproveButton
+                disabled={allowanceState !== ButtonStates.idle}
+                extraText
+                onClick={approve}
+                state={allowanceState}
+              >
+                {allowanceState === ButtonStates.idle &&
+                  `Approve ${currencySelected === ExchangeCurrency.Dai ? 'DAI' : 'OMN'}`}
+                {allowanceState === ButtonStates.working && 'Approving'}
+                {allowanceState === ButtonStates.finished && 'Approved'}
```

Comment:
after setting allowance the green Approved message shows up just fine but is quickly hidden again. I think we might need two seperate allowanceStates for omn and dai. Then for example we can update `daiWalletAllowance` and to be true if `daiAllowanceState` is `ButtonStates.finished` so that the Approved message can continue to be shown after setting allowance.

Reply:
great feedback and can confirm regarding the quickly hidden thing.